README.md

```diff
@@ -15,17 +15,62 @@ Implement the tool `st` that does exact pattern matching using a suffix tree. Te
 
 ## Specify if you have used a linear time or quadratic time algorithm.
 
+# ANWSER:
+I have implemented the naive-ST-algorithm which is build with O(n^2)
+and searched with O(m).
+
 ## Insights you may have had while implementing and comparing the algorithms.
 
+# ANWSER:
+That suffix trees are a magical and awesome.
+
 ## Problems encountered if any.
 
-## Correctness
+# ANWSER:
+That implementing something which output is hard to visualize (while 
+building/implementing) is extremely challenging. 
```

Comment:
Yes, you need to consider how to test upfront and add testing code everywhere. You might not be able to visualise the tree, but your program can self-test if you tell it how to.